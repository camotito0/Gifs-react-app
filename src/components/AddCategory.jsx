import { useState } from "react"

export const AddCategory = ({	onNewCategory	}) => {

	const [inputValue, setInputValue] = useState('')

	const onInputChange = (event) => {
		setInputValue(event.target.value)
	}

	const handleInput = (e) => {
		e.preventDefault()
		if(inputValue.trim().length <= 1) return ;
		onNewCategory(inputValue.trim())
		setInputValue('')
	}

	return (
		<form onSubmit= { handleInput }>
			<input
				type = 'text'
				placeholder = 'Buscar gifs'
				value = { inputValue }
				onChange = { onInputChange }
			/>
		</form>
	)
}