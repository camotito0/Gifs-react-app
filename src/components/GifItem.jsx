import PropTypes from 'prop-types';

export const GifItem = ({ name, url }) => {
	return (
		<div className="card">
			<img src = { url } alt = { name }/>
			<p>{ name	}</p>
		</div>
	)
}

// Agregamos ttypes a los parametros y 
// tambien le decimos que son requereidos
GifItem.propTypes = {
  name: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
}