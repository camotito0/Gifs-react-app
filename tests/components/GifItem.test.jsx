import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components';

describe('Tests for the component <GifItem />', () => {
	const name = 'Valorant';
	const url =  'https://media2.giphy.com/media/yQQMjKgLhRc9Y8stQV/giphy.gif?cid=00210a5c9z1tqnlknq84zqdii1yis6kt8x9zz34hfcuu8uf2&rid=giphy.gif&ct=g'

	it('Should match the snapshot with component', () => {
		const {	container	} =	render(<GifItem name={name} url={url} />);
		expect( container ).toMatchSnapshot();
	});

	it('the tag img most contain the url and alt', () => {
		render(<GifItem name={name} url={url} />);
		const {	src, alt } = screen.getByRole('img');
		expect(src).toBe(url);
		expect(alt).toBe(name);
	});

	it('should render the title in a p tag', () => {
		render(<GifItem name={name} url={url} />);
		/* screen.debug(); */
		expect(screen.getByText(name)).toBeTruthy();
	});
});