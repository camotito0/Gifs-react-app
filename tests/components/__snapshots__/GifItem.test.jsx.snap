// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for the component <GifItem /> Should match the snapshot with component 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Valorant"
      src="https://media2.giphy.com/media/yQQMjKgLhRc9Y8stQV/giphy.gif?cid=00210a5c9z1tqnlknq84zqdii1yis6kt8x9zz34hfcuu8uf2&rid=giphy.gif&ct=g"
    />
    <p>
      Valorant
    </p>
  </div>
</div>
`;
